File: src/algorithms/two-out-of-three.js
/**
 * Given three integer arrays, returns a distinct array containing all the
 * values that are present in at least two out of the three arrays.
 * 
 * @param {number[]} nums1 - First array of numbers
 * @param {number[]} nums2 - Second array of numbers
 * @param {number[]} nums3 - Third array of numbers
 * @returns {number[]} - Returns an array of distinct values
 */
function twoOutOfThree (nums1, nums2, nums3) {
  let res = new Set();
  let set1 = new Set(nums1);
  let set2 = new Set(nums2);
  let set3 = new Set(nums3);
  set1.forEach(element => {
    if (set2.has(element) || set3.has(element)) {
      res.add(element);
    }
  });
  set2.forEach(element => {
    if (set3.has(element)) {
      res.add(element);
    }
  });
  return Array.from(res);
}

File: src/algorithms/two-out-of-three.py
def two_out_of_three(nums1, nums2, nums3):
    """Given three integer arrays nums1, nums2, and nums3, return a distinct 
    array containing all the values that are present in at least two out of the 
    three arrays. You may return the values in any order.

    Args:
        nums1 (list[int]): A list of positive integers.
        nums2 (list[int]): A list of positive integers.
        nums3 (list[int]): A list of positive integers.

    Returns:
        list[int]: Returns a list of integers that occur in at least two lists.
    """
    res = []
    s1 = set(nums1)
    s2 = set(nums2)
    s3 = set(nums3)
    for num in s1:
        if num in s2 or num in s3:
            res.append(num)
    for num in s2:
        if num in s3 and num not in res:
            res.append(num)
    return res
